src/simsopt/_core/graph_optimizable.py

```diff
@@ -582,8 +583,10 @@ def __str__(self):
         return self.name
 
     def __hash__(self) -> int:
-        hash_str = hashlib.sha256(self.name.encode('utf-8')).hexdigest()
```

Comment:
I'd compute the hash in the __init__ and store it as _hash and return it when __hash__ is called. Eliminates one check